City-of-Helsinki/berth-reservations-admin
# VEN-904 | Enable disconnecting customers from applications

## Description :sparkles:

Adds button to disconnect a customer from an application in `PENDING` state.

Additionally refactors all application views to use a common
header component.

### Closes :no_good_woman:

[VEN-904](https://helsinkisolutionoffice.atlassian.net/browse/VEN-904)

## Testing :alembic:

### Automated tests :gear:️

- `src/common/applicationHeader/__tests__/ApplicationHeader.test.tsx`
- Added test for `canUnlinkCustomer` in `src/common/utils/__tests__/applicationUtils.test.ts`
- Updated other relevant tests.

### Manual testing :construction_worker_man:

- Open an application in state "odottaa käsittelyä". If its unconnected, connect it to a customer.
- The button "Poista asiakasliitos" should be visible
- After clicking and confirming, the customer selection table should be present for you to reconnect the application

## Additional notes :spiral_notepad:

Pending on https://github.com/City-of-Helsinki/berth-reservations/pull/329.
